src/sentry/integrations/slack/views/link_team.py

```diff
@@ -102,26 +99,13 @@ def handle(self, request: Request, signed_params: str) -> HttpResponse:
             return render_error_page(request, body_text="HTTP 403: Invalid team ID")
 
         try:
-            identity = Identity.objects.select_related("user").get(
-                idp=idp, external_id=params["slack_id"]
-            )
+            Identity.objects.select_related("user").get(idp=idp, external_id=params["slack_id"])
```

Comment:
Now that we're not getting the `identity`, we don't need the `select_related` and the try/catch.
```python
if not Identity.objects.filter(idp=idp, external_id=params["slack_id"]).exists():
  ...
```